packages/components/src/Heading/Heading.tsx

```diff
@@ -0,0 +1,44 @@
+import { TextProps, TypographyComponent } from "../Text/Text";
+import React from "react";
+import { styled } from "twin.macro";
+
+type HeadingElement = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+
+export type HeadingProps = TextProps & {
+  /**
+   * This prop can be used to specify which size heading should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the styling.
+   */
+  as?: HeadingElement;
+};
+
+const HeadingComponent = styled(TypographyComponent).attrs({
+  bold: true,
+})``;
+
+function Heading(props: HeadingProps): JSX.Element {
+  const {
+    as,
+    children,
+    color = "base",
+    size,
+    /**
+     * Components that wrap typography sometimes requires props such as event handlers
+     * to be passed down into the element. One example is the tooltip component.  It
+     * attaches a onHover and onFocus event to the element to determine when to
+     * trigger the overlay.
+     */
+    ...rest
```

Comment:
So this will allow passing `className`, which I'm cool with 😄 . But wanted to point that out

Reply:
hah you're right 🤭